ToNote/Views/NoteView.xaml.cs

```diff
@@ -43,5 +44,16 @@ public NoteView()
                 e.Handled = true;
             };
         }
+        private void Button1_Click(object sender, System.Windows.RoutedEventArgs e)
+        {
+            Note note = (Note)DataContext;
+
+            var tagName = ((Button)sender).DataContext.ToString();
```

Comment:
All of this breaks MVC. This should be done with commands, like adding tags and everything else. Consult @t-eliks on how to if you get stuck.